src/main/scala/com/databricks/labs/mosaic/expressions/base/RasterBandExpression.scala

```diff
@@ -0,0 +1,59 @@
+package com.databricks.labs.mosaic.expressions.base
+
+import com.databricks.labs.mosaic.core.geometry.api.GeometryAPI
+import com.databricks.labs.mosaic.core.raster.{MosaicRaster, MosaicRasterBand}
+import com.databricks.labs.mosaic.core.raster.api.RasterAPI
+import com.databricks.labs.mosaic.functions.MosaicContext
+import org.apache.spark.sql.catalyst.expressions.{Expression, NullIntolerant, TernaryExpression}
+import org.apache.spark.sql.types.DataType
+import org.apache.spark.unsafe.types.UTF8String
+
+import scala.reflect.ClassTag
+
+abstract class RasterBandExpression[T <: Expression: ClassTag](
+    rasterExpr: Expression,
+    bandExpr: Expression,
+    pathExpr: Expression,
+    outputType: DataType
+) extends TernaryExpression
+      with NullIntolerant
+      with Serializable {
+
+    protected val geometryAPI: GeometryAPI = MosaicContext.geometryAPI
+
+    protected val rasterAPI: RasterAPI = MosaicContext.rasterAPI
+
+    override def first: Expression = rasterExpr
+
+    override def second: Expression = bandExpr
+
+    override def third: Expression = pathExpr
+
+    /** Output Data Type */
+    override def dataType: DataType = outputType
+
+    def bandTransform(raster: MosaicRaster, band: MosaicRasterBand): Any
+
+    override def nullSafeEval(inputRaster: Any, inputBand: Any, inputPath: Any): Any = {
```

Comment:
`inputBand` is actually the band index, and `inputPath` is probably a temp path for buffering?
Could you please add a docstring to explain the parameters and rename inputBand to bandIndex?

Reply:
will add docs for all methods.